src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8181,52 +8181,302 @@ public class Test : ITest
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : [|IInterface<string>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string>
+{{
+    public string MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    public string MethodDisallowNull(string value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNullableReferenceType()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : [|IInterface<string?>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string?>
+{{
+    public string? MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    [return: NotNull]
+    public string? MethodDisallowNull([DisallowNull] string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        // class Class<T> : IInterface<T?> where T : class
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNonNullableReferenceTypeConstraint()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class<T> : [|IInterface<T?>|] where T : class
+{{
+}}",
+$@"#nullable enable
 
 using System.Diagnostics.CodeAnalysis;
 
 {NullableAttributesCode}
 
-interface IFoo<T>
+interface IInterface<T>
 {{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class<T> : IInterface<T?> where T : class
+{{
+    public T MethodAllowNull(T value)
```

Comment:
❕ The signature here is not correct

```suggestion
    public T? MethodAllowNull(T? value)
```